### PR TITLE
Pass kwargs to subsequent queries in GitlabList (fixes: #1123)

### PR DESCRIPTION
**Pull request by tyates-indeed**
Pass kwargs to subsequent queries in GitlabList (fixes: #1123)

### CODE REVIEW

File: gitlab/__init__.py [hunk 1]
```diff
@@ -780,6 +780,12 @@ def __init__(self, gl, url, query_data, get_next=True, **kwargs):
         self._query(url, query_data, **kwargs)
         self._get_next = get_next
 
+        # Preserve kwargs for subsequent queries
```
Comment (rkpatel7) [inline gitlab/__init__.py:783]: Should this happen before https://github.com/python-gitlab/python-gitlab/pull/1124/commits/1d011ac72aeb18b5f31d10e42ffb49cf703c3e3a#diff-15f1286708cbbf354eafe7b7f49a1622L780? And at that line do:

```python
self._query(url, query_data, **self._kwargs)
```

That way this class would be internally consistent with how it calls `self._query()` in both `__init__()` and `next()`.

File: gitlab/__init__.py [hunk 2]
```diff
@@ -780,6 +780,12 @@ def __init__(self, gl, url, query_data, get_next=True, **kwargs):
...
+        if kwargs is None:
+            self._kwargs = {}
+        else:
+            self._kwargs = kwargs.copy()
+
     def _query(self, url, query_data=None, **kwargs):
```
[the remark this answers is not in the document]
Comment (tyates-indeed) [reply gitlab/__init__.py:790]: Good call. Changed.

File: gitlab/__init__.py (hunk 1 above)
Comment (tyates-indeed) [reply gitlab/__init__.py:781]: Yeah, that was my thinking as well.

File: gitlab/__init__.py [hunk 3]
```diff
@@ -777,7 +777,14 @@ class GitlabList(object):
 
     def __init__(self, gl, url, query_data, get_next=True, **kwargs):
         self._gl = gl
-        self._query(url, query_data, **kwargs)
+
+        # Preserve kwargs for subsequent queries
+        if kwargs is None:
+            self._kwargs = {}
+        else:
+            self._kwargs = kwargs.copy()
```
Comment (tron1point0) [inline gitlab/__init__.py:785]: `kwargs` can't be `None`:

```python
>>> def foo(**kwargs):
...   print(kwargs)
... 
>>> foo()
{}
>>> foo(**None)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    foo(**None)
TypeError: foo() argument after ** must be a mapping, not NoneType
```

So this  can just be:

```suggestion
        self._kwargs = kwargs.copy()
```

**Comment by rkpatel7** [comment]
@nejch, @max-wittig, or @bufferoverflow, any chance one of y'all can take a look at this pull request and merge it in?

**Comment by max-wittig** [comment]
Hi @rkpatel7 Sorry for the delay! I will have a look.

**Comment by max-wittig** [comment]
Thanks for the PR!

**Comment by tyates-indeed** [comment]
@max-wittig is there a release planned soon to pull in this change?

**Comment by max-wittig** [comment]
Sure makes sense

**Pull request merged**